src/IntegrationTests.jl

```diff
@@ -1,5 +1,159 @@
+        project_tree::AbstractDict=PkgDependency.builddict(Pkg.project().uuid, Pkg.project())
+    ) -> Vector{String}
+
+Returns a list of packages that have the package `package_name` as a dependency. 
+
+# Arguments
+
+- `package_name`: Name of the dependency
+- `package_filter`: Ignore all packages that do not match `package_filter`. This includes the 
+        top node package of the graph. Child nodes always are checked for `package_name`, but 
+        they are not traveres if they do not match `package_filter`.
+- `project_tree`: Project tree, where to search the dependent packages. Each (sub-) package 
+        needs to be AbstractDict{String, AbstractDict}
+
+# Returns
+
+all packages which have the searched dependency
+
+"""
+function depending_projects(
+    package_name::AbstractString,
+    package_filter::Union{TString,Regex},
+    project_tree::AbstractDict=PkgDependency.builddict(Pkg.project().uuid, Pkg.project()),
+)::Vector{String} where {TString<:AbstractString}
+    packages::Vector{String} = []
+    visited_packages::Vector{String} = []
+    _traverse_tree!(package_name, package_filter, project_tree, packages, visited_packages)
+    return packages
+end
+
+"""
+    depending_projects(
+        package_name::AbstractString, 
+        package_filter::AbstractVector{TString} where {TString<:AbstractString}, 
+        project_tree::AbstractDict=PkgDependency.builddict(Pkg.project().uuid, Pkg.project())
+    ) -> Vector{String}
+
+Returns a list of packages that have the package `package_name` as a dependency. 
+
+# Arguments
+    
+- `package_name`: Name of the dependency
+- `package_filter`: Ignore all packages that do not match `package_filter`. This includes the 
+        top node package of the graph. Child nodes always are checked for `package_name`, but 
+        they are not traveres if they do not match `package_filter`. Must be not empty.
+- `project_tree`: Project tree, where to search the dependent packages. Each (sub-) package 
+        needs to be AbstractDict{String, AbstractDict}
+    
+!!! note 
+    If you want to use a combination of srings and regexes in a vector, you should use a pure 
+    regex instead.
+
+# Returns
+
+all packages which have the searched dependency
+
+"""
+function depending_projects(
+    package_name::AbstractString,
+    package_filter::AbstractVector{TString},
+    project_tree::AbstractDict=PkgDependency.builddict(Pkg.project().uuid, Pkg.project()),
+)::Vector{String} where {TString<:AbstractString}
+    packages::Vector{String} = []
+    visited_packages::Vector{String} = []
+
+    if length(package_filter) == 0
+        throw(ArgumentError("package_filter must not be empty"))
+    else
+        regex_string = ""
+        for (i, s) in enumerate(package_filter)
+            if i == length(package_filter)
+                regex_string *= "^$(s)\$"
+            else
+                regex_string *= "^$(s)\$|"
+            end
+        end
+        reg = Regex(regex_string)
+    end
```

Comment:
```suggestion
    length(package_filter) == 0 || throw(ArgumentError("package_filter must not be empty"))
    
    regex_string = ""
    for (i, s) in enumerate(package_filter)
        if i == length(package_filter)
            regex_string *= "^$(s)\$"
        else
            regex_string *= "^$(s)\$|"
        end
    end
    reg = Regex(regex_string)
```

Reply:
Another thing, is this way of interpolating the strings into a regex safe? Or should certain characters be escaped properly?